homeassistant/components/sms/notify.py

```diff
@@ -19,8 +19,18 @@
 
 def get_service(hass, config, discovery_info=None):
     """Get the SMS notification service."""
-    gateway = hass.data[DOMAIN]
-    number = config[CONF_RECIPIENT]
+
+    if SMS_GATEWAY not in hass.data[DOMAIN]:
+        _LOGGER.error("SMS gateway not found, cannot initialize service")
+        return
```

Comment:
We could start the config flow from here to notify the user that a config entry is needed.

Reply:
Can you be more specific? What do you mean by start the config flow?

Could you point to some sample code?

Reply:
I've thought some more about this and I think we don't need to start the flow here. We already start the flow with import source in `async_setup` if config yaml is present. That should create the entry before this platform is setup.

We can keep this guard, just in case there's some path we haven't thought about, or if there's some race condition.